File: Code Forces and Leetcode2/E_Sort_the_Array.py
n = int(input())
a = list(map(int, input().split()))

start, end = -1, -1

for i in range(n - 1):
    if a[i] > a[i + 1]:
        start = i
        break

if start == -1:
    print("yes\n1 1")
    exit(0)

for i in range(n - 1, start, -1):
    if a[i] < a[i - 1]:
        end = i
        break

a[start:end + 1] = reversed(a[start:end + 1])

if any(a[i] > a[i + 1] for i in range(n - 1)):
    print("no")
else:
    print("yes")
    print(start + 1, end + 1)

File: Code Forces and Leetcode2/E_Sort_the_Array.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> a(n);

    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }

    int start = -1, end = -1;

    // Find the first element that is not in increasing order
    for (int i = 0; i < n - 1; ++i) {
        if (a[i] > a[i + 1]) {
            start = i;
            break;
        }
    }

    // If the array is already sorted, print "yes" and indices (1 1)
    if (start == -1) {
        cout << "yes\n1 1\n";
        return 0;
    }

    // Find the last element that is not in increasing order
    for (int i = n - 1; i > start; --i) {
        if (a[i] < a[i - 1]) {
            end = i;
            break;
        }
    }

    // Reverse the segment and check if the array becomes sorted
    reverse(a.begin() + start, a.begin() + end + 1);

    // Check if the array is sorted after reversing the segment
    for (int i = 0; i < n - 1; ++i) {
        if (a[i] > a[i + 1]) {
            cout << "no\n";
            return 0;
        }
    }

    // Print "yes" and the indices of the reversed segment
    cout << "yes\n" << start + 1 << " " << end + 1 << "\n";

    return 0;
}
